### PR TITLE
Allow local run in k8s by port forwarding

### PR DESCRIPTION
**Pull request by materkey**
Expected:
When I want to debug runner I run it locally
Actual:
K8s network isolation does not allow me to access devices

### CODE REVIEW

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/config/RunnerInputParams.kt [hunk 1]
```diff
@@ -32,6 +32,7 @@ public data class RunnerInputParams(
     val saveTestArtifactsToOutputs: Boolean,
     val useLegacyExtensionsV1Beta: Boolean,
     val sendPodsMetrics: Boolean,
+    val needForward: Boolean,
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/config/RunnerInputParams.kt:35]: Intention is not clear

**Comment by sergioboi** [comment]
Could you describe why LoocalDeviceProvider doesn't fit your goals?

File: subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/RemoteDeviceProvider.kt [hunk 2]
```diff
@@ -65,15 +65,19 @@ internal class RemoteDeviceProviderImpl(
             val podIp = ip
             requireNotNull(podIp) { "Pod: $name must have an IP" }
 
-            val serial = emulatorSerialName(name = podIp)
+            val name = portForward?.let { portForward ->
```
Comment (sergioboi) [inline subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/RemoteDeviceProvider.kt:68]: I'm defenetly think that If this change would be done it should be done with new layer of absctraction. For example `DevicePortProvider`

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt [hunk 3]
```diff
@@ -205,6 +206,9 @@ public abstract class InstrumentationTestsTask @Inject constructor(
             saveTestArtifactsToOutputs = experiments.saveTestArtifactsToOutputs,
             useLegacyExtensionsV1Beta = experiments.useLegacyExtensionsV1Beta,
             sendPodsMetrics = experiments.sendPodsMetrics,
+            needForward = project.getOptionalStringProperty(
```
Comment (sergioboi) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt:209]: You should add this to gradle Extention. And choose better name. `adbDevicePortForward` for example

File: subprojects/test-runner/kubernetes/src/main/kotlin/com/avito/k8s/model/KubePod.kt [hunk 4]
```diff
@@ -3,11 +3,12 @@ package com.avito.k8s.model
 import io.fabric8.kubernetes.api.model.ContainerStatus
 import io.fabric8.kubernetes.api.model.Pod
 import io.fabric8.kubernetes.api.model.PodStatus
+import io.fabric8.kubernetes.client.LocalPortForward
 
 /**
  * Wrapper to deal with nullability and abstract from fabric8io API changes
  */
-public class KubePod(private val pod: Pod) {
+public class KubePod(private val pod: Pod, public val portForward: LocalPortForward? = null) {
```
Comment (sergioboi) [inline subprojects/test-runner/kubernetes/src/main/kotlin/com/avito/k8s/model/KubePod.kt:11]: should be resolved with inheritence. I don't like optional fields they smell

File: subprojects/test-runner/kubernetes/src/main/kotlin/com/avito/k8s/KubernetesApiImpl.kt [hunk 5]
```diff
@@ -13,6 +13,7 @@ import kotlinx.coroutines.delay
 internal class KubernetesApiImpl(
     private val kubernetesClient: KubernetesClient,
     loggerFactory: LoggerFactory,
+    override val needForward: Boolean,
```
Comment (sergioboi) [inline subprojects/test-runner/kubernetes/src/main/kotlin/com/avito/k8s/KubernetesApiImpl.kt:16]: We should change bool to sealed class PodPortForwardConfig { disabled, Enabled(port = 5555)

**Review by sergioboi — CHANGES_REQUESTED** [review]

**Comment by dsvoronin** [comment]
Thanks for the contribution, but we decided to halt development of the test-runner in favor of "EmCee for android"
This PR will not be merged.

You can fork it if you want

**Pull request closed without merging**